xml_converter/src/packaging_protobin.cpp

```diff
@@ -284,3 +291,47 @@ void write_protobuf_file_per_map_id(
             &state);
     }
 }
+void _category_filter_by_depth(
+    const map<string, Category>* marker_categories,
+    int target_depth,
+    map<string, StringHierarchy>* category_hierarchy,
+    vector<string> parent_name,
+    int current_depth = 0) {
+    for (auto it = marker_categories->begin(); it != marker_categories->end(); it++) {
+        vector<string> child_name = parent_name;
+        child_name.push_back(it->first);
+        if (target_depth == current_depth) {
+            StringHierarchy category_filter;
+            category_filter.add_path(child_name, true);
+            (*category_hierarchy)[join(child_name, "_")] = category_filter;
```

Comment:
If `it->first` can contain an `_` then this could cause a collision in category_hierarchy. If we had a structure like
```
one_two
  three
one
  two_three
```

That would be a valid tree with no duplicates. But both of those would end up producing the key `one_two_three` here and collide.

Reply:
I had just defaulted to snake case. I'm going to change this to `.` since i know that should not be in the name field since we are using it as delimiter to determine hierarchy. 

The name field is normalized when it is written proto->xml so perhaps we should look at normalizing the xml->xml as well.